ts/Dashboards/Component/Component.ts

```diff
@@ -518,6 +569,18 @@ abstract class Component<TEventObject extends Component.EventTypes = Component.E
 
     public setTitle(titleOptions: Component.TextOptionsType): void {
         const previousTitle = this.titleElement;
+
+        if (
+            !titleOptions || typeof titleOptions === 'string' ?
```

Comment:
You are using it in two places, each has different formatting.
How about extracting that logic somewhere outside?

Reply:
Think this will have to be for the backlog